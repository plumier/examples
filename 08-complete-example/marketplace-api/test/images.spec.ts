import { join } from "path"
import supertest from "supertest"

import createApp from "../src/app"
import { createUser } from "./_helper"


describe("Shops Images", () => {
    it("Should upload file properly", async () => {
        const app = await createApp({ mode: "production" })
        const user = await createUser(app)
        const { body: img } = await supertest(app.callback())
            .post(`/api/images/upload`)
            .attach("file", join(__dirname, "image/landscape.jpeg"))
            .set("Authorization", `Bearer ${user.token}`)
            .expect(200)
        expect(img).toMatchSnapshot()
    })
})