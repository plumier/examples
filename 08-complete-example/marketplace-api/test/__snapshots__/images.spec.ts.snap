// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shops Images Should upload file properly 1`] = `
Object {
  "id": 1,
}
`;
